### PR TITLE
Add support for initial cell selection

### PR DESCRIPTION
**Pull request by nojaf**
## 📝 Summary

Another follow-up on https://github.com/marimo-team/marimo/pull/3725, this time to add support for initial cell selection.

## 🔍 Description of Changes

`initial_selection` can also be a list of dict like `initial_selection = [{"rowId":"0", "columnName":"metric2"}, {"rowId":"1", "columnName":"id"}]` after this PR.

## 📋 Checklist

- [x] I have read the [contributor guidelines](https://github.com/marimo-team/marimo/blob/main/CONTRIBUTING.md).
- [ ] For large changes, or changes that affect the public API: this change was discussed or approved through an issue, on [Discord](https://marimo.io/discord?ref=pr), or the community [discussions](https://github.com/marimo-team/marimo/discussions) (Please provide a link if applicable).
- [ ] I have added tests for the changes made.
- [x] I have run the code and verified that it works as expected.

## 📜 Reviewers

@mscolnick

Let me know if this needs further tweaks.

### CODE REVIEW

File: marimo/_plugins/ui/_impl/table.py [hunk 1]
```diff
@@ -322,20 +324,46 @@ def __init__(
...
+                    self._selected_manager = (
+                        self._searched_manager.select_cells(
+                            [
+                                TableCoordinate(
+                                    row_id=v["rowId"],
+                                    column_name=v["columnName"],
+                                )
+                                for v in initial_selection
+                                if isinstance(v, dict)
+                                and "rowId" in v
```
Comment (mscolnick) [inline marimo/_plugins/ui/_impl/table.py:352]: i think this might be better as a tuple instead of `rowId` and `columnId`. python developers use `snake_case` as well so it wouldn't be intuitive for them to construct dictionaries with `rowId/columnName`

**Comment by mscolnick** [comment]
this approach looks great, i think next steps would be some tests for happy-path, edge-cases, and failure cases (in correct types passed, downloading cell selection, etc)

File: marimo/_plugins/ui/_impl/table.py (hunk 1 above)
Comment (nojaf) [reply marimo/_plugins/ui/_impl/table.py:352]: So something like
```py
        initial_selection: Optional[
            Union[list[int], list[tuple[str, str]]]
        ] = None,
```

File: marimo/_plugins/ui/_impl/table.py [hunk 2]
```diff
@@ -322,25 +324,57 @@ def __init__(
...
+                            ]
+                        )
+                    )
             except IndexError as e:
                 raise IndexError(
                     "initial_selection contains invalid row indices"
                 ) from e
-            initial_value = initial_selection
+
+            for v in initial_selection:
```
Comment (nojaf) [inline marimo/_plugins/ui/_impl/table.py:370]: @mscolnick how do I make py-check happy here?

Comment (mscolnick) [reply marimo/_plugins/ui/_impl/table.py:370]: you could either case or `# type: ignore[reason]`

**Review by mscolnick — APPROVED** [review]
this looks great, nice work!
